hoomd/util.py

```diff
@@ -248,3 +248,27 @@ def array_to_strings(value):
         return string_list
     else:
         return value
+
+
+def create_NotImplementedClass(error, reason):
```

Comment:
This seems overengineered to me. What's the reason to not just define the NoGPU class with a simple `__init__` that throws a `GPUNotAvailableError`? Metaclasses shouldn't be used unless they're really needed IMO, they make it much harder to read the code for most people.

Reply:
That is a fair critique. I was thinking there may be some use in the future, but it is unlikely and easier just to hard code another case.